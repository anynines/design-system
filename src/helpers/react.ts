import React from 'react'

export const onClickOutsideHook = (
  ref: React.RefObject<HTMLDivElement> | null,
  callback: () => void
): void => {
  // Hook get from https://stackoverflow.com/a/42234988/8583669

  // eslint-disable-next-line react-hooks/rules-of-hooks
  React.useEffect(() => {
    const handleClickOutside = (event: MouseEvent): void => {
      if (ref?.current && !(ref.current.contains(event.target as Node))) {
        callback()
      }
    }
    // Bind the event listener
    document.addEventListener('mousedown', handleClickOutside)
    return (): void => {
      // Unbind the event listener on clean up
      document.removeEventListener('mousedown', handleClickOutside)
    }
  }, [callback, ref])
}
