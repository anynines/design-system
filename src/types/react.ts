import { AriaAttributes, DOMAttributes } from 'react'

export type ReactHtmlElementProps<T extends AriaAttributes | DOMAttributes<T>>
  = React.DetailedHTMLProps<React.HTMLAttributes<T>, T>

export type DefaultComponentProps = Pick<ReactHtmlElementProps<HTMLDivElement>, 'className' | 'style'>
