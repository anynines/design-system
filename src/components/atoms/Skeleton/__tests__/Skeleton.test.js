import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import TitleSkeleton from '../Title'
import CardSkeleton from '../Card'
import HeroSkeleton from '../Hero'
import LineSkeleton from '../Line'
import TextSkeleton from '../Text'
import WordSkeleton from '../Word'

const Skeletons = (
  <>
    <TitleSkeleton />
    <TitleSkeleton title='h3' />
    <CardSkeleton />
    <HeroSkeleton />
    <LineSkeleton />
    <LineSkeleton width='sm' />
    <LineSkeleton width='md' />
    <TextSkeleton />
    <TextSkeleton width='sm' />
    <TextSkeleton width='md' />
    <TextSkeleton lines={5} />
    <WordSkeleton />
  </>
)

describe('Skeletons component', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const component = render(Skeletons)
      const { asFragment } = component
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
