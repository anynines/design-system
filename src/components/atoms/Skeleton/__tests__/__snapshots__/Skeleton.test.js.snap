// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeletons component snapshot should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c0.width-lg {
  width: 100%;
}

.c0.width-md {
  width: 50%;
}

.c0.width-sm {
  width: 25%;
}

.c1 {
  position: absolute;
  top: 0;
}

<h1
    class="title-skeleton "
    style="width: 100%; position: relative;"
  >
      
    <span
      class="c0 line-skeleton c1 width-lg"
    />
  </h1>
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c0.width-lg {
  width: 100%;
}

.c0.width-md {
  width: 50%;
}

.c0.width-sm {
  width: 25%;
}

.c1 {
  position: absolute;
  top: 0;
}

<h3
    class="title-skeleton "
    style="width: 100%; position: relative;"
  >
      
    <span
      class="c0 line-skeleton c1 width-lg"
    />
  </h3>
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  margin-bottom: 20px;
  background-color: var(--text-color);
  width: 100%;
  height: 200px;
  border-radius: var(--radius);
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 500px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
}

<div
    class="c0 line-skeleton "
  />
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  margin-bottom: 20px;
  background-color: var(--text-color);
  width: 100%;
  height: 250px;
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 500px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
}

<div
    class="c0 line-skeleton "
  />
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c0.width-lg {
  width: 100%;
}

.c0.width-md {
  width: 50%;
}

.c0.width-sm {
  width: 25%;
}

<span
    class="c0 line-skeleton  width-lg"
  />
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c0.width-lg {
  width: 100%;
}

.c0.width-md {
  width: 50%;
}

.c0.width-sm {
  width: 25%;
}

<span
    class="c0 line-skeleton  width-sm"
  />
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c0.width-lg {
  width: 100%;
}

.c0.width-md {
  width: 50%;
}

.c0.width-sm {
  width: 25%;
}

<span
    class="c0 line-skeleton  width-md"
  />
  .c1 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c1:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c1.width-lg {
  width: 100%;
}

.c1.width-md {
  width: 50%;
}

.c1.width-sm {
  width: 25%;
}

.c0 {
  margin-bottom: 25px;
}

.c0 p {
  margin-top: 0;
  margin-bottom: 5px;
}

.c2 {
  position: absolute;
  top: 0;
  margin-bottom: 0;
}

<div
    class="c0 text-skeleton "
  >
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-lg"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-lg"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-sm"
      />
    </p>
  </div>
  .c1 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c1:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c1.width-lg {
  width: 100%;
}

.c1.width-md {
  width: 50%;
}

.c1.width-sm {
  width: 25%;
}

.c0 {
  margin-bottom: 25px;
}

.c0 p {
  margin-top: 0;
  margin-bottom: 5px;
}

.c2 {
  position: absolute;
  top: 0;
  margin-bottom: 0;
}

<div
    class="c0 text-skeleton "
  >
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-sm"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-sm"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-sm"
      />
    </p>
  </div>
  .c1 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c1:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c1.width-lg {
  width: 100%;
}

.c1.width-md {
  width: 50%;
}

.c1.width-sm {
  width: 25%;
}

.c0 {
  margin-bottom: 25px;
}

.c0 p {
  margin-top: 0;
  margin-bottom: 5px;
}

.c2 {
  position: absolute;
  top: 0;
  margin-bottom: 0;
}

<div
    class="c0 text-skeleton "
  >
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-md"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-md"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-sm"
      />
    </p>
  </div>
  .c1 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c1:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c1.width-lg {
  width: 100%;
}

.c1.width-md {
  width: 50%;
}

.c1.width-sm {
  width: 25%;
}

.c0 {
  margin-bottom: 25px;
}

.c0 p {
  margin-top: 0;
  margin-bottom: 5px;
}

.c2 {
  position: absolute;
  top: 0;
  margin-bottom: 0;
}

<div
    class="c0 text-skeleton "
  >
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-lg"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-lg"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-lg"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-lg"
      />
    </p>
    <p
      style="position: relative;"
    >
       
      <span
        class="c1 line-skeleton c2 width-sm"
      />
    </p>
  </div>
  .c0 {
  position: relative;
  overflow: hidden;
  opacity: .25;
  background-color: var(--text-color);
  height: 100%;
  border-radius: var(--radius);
}

.c0:before {
  position: absolute;
  top: 0;
  left: -150px;
  opacity: .75;
  background-image: linear-gradient( 0deg, rgba(230,230,230,0), var(--box-bg-color), rgba(230,230,230,0) );
  width: 200px;
  height: 50px;
  color: transparent;
  content: 'lorem ipsum';
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  -webkit-animation: shine 1.5s ease infinite;
  animation: shine 1.5s ease infinite;
  -webkit-animation-delay: 1s;
  animation-delay: 1s;
}

.c0.width-lg {
  width: 100%;
}

.c0.width-md {
  width: 50%;
}

.c0.width-sm {
  width: 25%;
}

.c1 {
  position: absolute;
  top: 0;
}

<span
    class=""
    style="position: relative;"
  >
      
    <span
      class="c0 line-skeleton c1 width-lg"
    />
  </span>
</DocumentFragment>
`;
